src/class_loader.cpp

```diff
@@ -58,17 +56,14 @@ std::string systemLibraryPrefix()
 
 std::string systemLibrarySuffix()
 {
-// Poco should be compiled with `#define POCO_NO_SHARED_LIBRARY_DEBUG_SUFFIX`
-// to automatically remove the trailing `d` from the shared library suffix
-//   return Poco::SharedLibrary::suffix();
 #ifdef __linux__
   return ".so";
 #elif __APPLE__
   return ".dylib";
 #elif _WIN32
   return ".dll";
 #else
-  return Poco::SharedLibrary::suffix();
+  return "";
```

Comment:
I'm curious what the poco function was doing before, maybe we should raise not implemented here if it's a platform we do not support?

Reply:
https://pocoproject.org/docs/Poco.SharedLibrary.html#18310

Reply:
Hmm, that documentation doesn't say what happens if the platform is not supported.

Looks like this error case is detected at compile time instead (it would fail to compile?):

https://github.com/pocoproject/poco/blob/ed3d164904b9e2d00efd46889251e44ccaf1c788/Foundation/src/SharedLibrary.cpp#L19-L27

In which case maybe our `#else` should just contain an `#error`?

Reply:
an error like this one:

```
#error there is no library suffix defined for this platform
```

instead of returning an empty error?

Reply:
I think so, yes. Or something like:

```c
#error Unsupported OS, dynamic library suffix is unknown.
```